encoding/jsonc_test.ts

```diff
@@ -171,3 +172,47 @@ Deno.test({
     assertValidParse('  ["foo\\\\\\\\", "bar"]', ["foo\\\\", "bar"]);
   },
 });
+
+Deno.test({
+  name: "[jsonc] avoid prototype pollution",
```

Comment:
Is it meaningful to run this test case in Deno?

Reply:
This was just a test case reinforcementation. (This is the code I used to check the current Deno behavior.)

I've left this code in case someone wants to see how Deno works now, but it's not that important. Should it be deleted?

Reply:
I think the below one (`[jsonc] use Object.defineProperty when setting object property`) is the only test necessary for this fix as this test case already passes without this PR's change (It checks Deno cli behavior instead of `std/encoding/jsonc` behavior in my view)

Reply:
It makes sense. I removed this test.

Reply:
Thank you for your review!